README.md

```diff
@@ -321,6 +321,36 @@ use Cake\Event\EventManager;
 EventManager::instance()->on(new SocialAuthListener());
 ```
 
+### Extend with custom providers
+
+In order to enable custom or not preconfigured provideres you can extend the middleware configuration with *serviceConfig.collectionFactory* and create your own instance of *SocialConnect\Auth\CollectionFactory*.
+
+Attach the listener in your `Application` class:
+
+```php
+// src/Application.php
+SocialConnect\Auth\CollectionFactory;
+
+// Instantiate your own CollectionFactory with
+...
+public function middleware($middlewareQueue): \Cake\Http\MiddlewareQueue
+...
+$cf = new CollectionFactory();
+$cf->register(\App\Authenticator\CustomProvider::NAME, \App\Authenticator\CustomProvider::class);
+...
+
+// Add this to your configuration
+serviceConfig.collectionFactory => $cf,
```

Comment:
Don't nest `collectionFactory` under `serviceConfig`. Keep it as a separate top level config. Additional key inside the service options argument could cause problems in future.

Reply:
yes, moved